cloudconfig/cloudinit/interface.go

```diff
@@ -489,18 +490,21 @@ func New(ser string) (CloudConfig, error) {
 	}
 }
 
+// TODO: de duplicate with cloudconfig/instancecfg
 // SSHKeys contains SSH host keys to configure on a machine.
-type SSHKeys struct {
-	RSA *SSHKey
-}
+type SSHKeys []SSHKey
 
+// TODO: de duplicate with cloudconfig/instancecfg
 // SSHKey is an SSH key pair.
 type SSHKey struct {
 	// Private is the SSH private key.
 	Private string
 
 	// Public is the SSH public key.
 	Public string
+
+	// PublicKeyType contains the public key type as defined by golang.org/x/crypto/ssh KeyAlgo*
+	PublicKeyType string
```

Comment:
Would this be better named to PublicKeyAlgorithm to match what it's being set to?